code/__HELPERS/mobs.dm

```diff
@@ -499,22 +499,31 @@ Proc for attack log creation, because really why not
 
 //Used in chemical_mob_spawn. Generates a random mob based on a given gold_core_spawnable value.
 /proc/create_random_mob(spawn_location, mob_class = HOSTILE_SPAWN)
-	var/static/list/mob_spawn_meancritters = list() // list of possible hostile mobs
-	var/static/list/mob_spawn_nicecritters = list() // and possible friendly mobs
-
-	if(mob_spawn_meancritters.len <= 0 || mob_spawn_nicecritters.len <= 0)
+	var/static/list/mob_spawn_meancritters // list of possible hostile mobs
+	var/static/list/mob_spawn_nicecritters // and possible friendly mobs
+	var/static/list/mob_spawn_neutralcritters // list of possible neutral mobs
+
+	if(!mob_spawn_meancritters)

```

Comment:
you should check for all three lists

Reply:
Okay, amended this as well, I think. Let me know if it's wrong.

Reply:
there is, L.len will throw a runtime if L is null.

Reply:
I specifically told him to do one because there will never be a situation where one will be null and the others will not be, because they are all initialized at the same time.

I told him to not do the .len thing because if one of the lists has a length of zero (as the neutral list currently will), it will cause the length of each list to double every time this proc is called, eventually causing the server to run out of memory and crash.

Reply:
:up:

Reply:
Alright, amended back to previous version.

Reply:
Lists are objects, they will always return true when tested; this check does nothing.

Reply:
They will not return true before they are initialized, which is the whole point of initializing them in the if.

Reply:
Will the list actually be static when you put it in a proc or will it generate a new list everytime this proc is called.

Reply:
The list is actually static but scoped to only this proc.